### ISSUE TITLE
FlashList accessibility confusing order

### ISSUE THREAD

**Issue by sharabai** [Expensify/App#86126]
## Action Performed:
Open any page that uses `FlashList`. Any component that uses `SelectionListWithSections` has it. 
For example:
1. go to Inbox. 
2. use Screen reader to navigate the list up and down

## Expected Result:
You can scroll all the way down and up

## Actual Result:
You are going to skip elements along the way and not be able to go all the way down and it will exit the list too early like it already finished displaying it.

## Workaround:
Use scrolling, like mouse scrolling. Apparently you can tell screen readers to perform scroll.
## Area issue was found in:
From my experience working on migrating `SelectionListWithSections`

## Failed WCAG checkpoints
Primary violation:
* 4.1.2: Name, Role, Value (A)
* 1.3.1: Info and Relationships (A)
* 2.4.3: Focus Order (A)

## User impact:
User can't navigate through most of the lists in our app reliably

## Suggested resolution:
Working on it in [this PR](https://github.com/Expensify/App/pull/85825) which focuses on patching FlashList so that its has the correct DOM order of its elements. 

## Platforms:

Which of our officially supported platforms is this issue occurring on? Please only tick the box if you have provided a screen-recording in the thread for each platform:
- [ ] Android: App
- [ ] Android: mWeb Chrome
- [ ] iOS: App
- [ ] iOS: mWeb Safari
- [ ] iOS: mWeb Chrome
- [x] Windows: Chrome
- [x] MacOS: Chrome / Safari

## Screenshots/Videos

**Version Number:** NA
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** NA
**Notes/Photos/Videos:** 

https://github.com/user-attachments/assets/2bb2a307-fcf7-41ce-8135-2d332b3f1084

**Expensify/Expensify Issue URL:** NA
**Issue reported by:** NA
**Slack conversation:** NA

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @sharabai</details>

**Comment by mitarachim** [comment]
Deploy Blocker https://github.com/Expensify/App/issues/91059 was identified to be related to the PR https://github.com/Expensify/App/pull/85825 that attempted to fix this issue.

**Comment by mallenexpensify** [comment]
@huult assigned to you, 👀 above  
> If there are posts on the PR or issue that reference a regression or deploy blocker, C+ posts on the issue and confirms regression (which will reduce pay by 50%) or provides reasoning why they shouldn’t be considered.  If the C+ doesn’t update the issue, the contributor manager making the payment will assume any mention of a regression or deploy blockers reduces the issue amount.

**Comment by huult** [comment]
@mallenexpensify The PR was reverted, and we are working on https://github.com/Expensify/App/pull/91328 to update the fix.

**Comment by huult** [comment]
This ticket has two PRs:
https://github.com/Expensify/App/pull/91328 -> merged
https://github.com/Expensify/App/pull/91605 -> still open

I believe payment should only be processed after the second PR is merged.

**Comment by huult** [comment]
The second PR is under review.

**Comment by huult** [comment]
The second [PR](https://github.com/Expensify/App/pull/91605) is under review.

**Comment by huult** [comment]
@mallenexpensify This ticket requires two PRs to be fully completed.

The first PR, https://github.com/Expensify/App/pull/85825, was merged initially but later reverted due to a regression test failure. The follow-up PR, https://github.com/Expensify/App/pull/91328, was opened to update and address that issue.

The second PR, https://github.com/Expensify/App/pull/91605, has been released to production. The payment for this ticket is expected on **06/07/2026**.

**Comment by huult** [comment]
# BugZero Checklist:

- [x] **[Contributor]** The offending PR and associated issue have been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

It looks like a new feature.

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:


-

### Test:


1. Go to Inbox.
2. Use Screen reader to navigate the list up and down
3. Use JAWS on Windows or VoiceOver on MacOS to move around in the list
4. Go forward so the list goes over the buffer and (if you can see 20 elements, scroll to element # 40)
5. Go back to start
6. It should follow natural order of items in the list

Do we agree 👍 or 👎

**Comment by huult** [comment]
This issue is ready for payment.

**Comment by mallenexpensify** [comment]
### Payment Summary
Contributor+: @huult  due $375 via NewDot

Regression, so pay is 50%  
^ Test case created. Thx

**Comment by huult** [comment]
@mallenexpensify We have two PRs, so I think the total would be $375 or $250. PR [#1](https://github.com/Expensify/App/pull/85825) would be 50% due to the regression testing, and PR [#2](https://github.com/Expensify/App/pull/91605) would be 100% as normal ($250 or $125)

**Comment by huult** [comment]
@mallenexpensify What do you think about this case?

**Comment by mallenexpensify** [comment]
@huult that makes sense to me.  increased price. cc @chuckdries in case you disagree
